src/services/merge-checker.ts

```diff
@@ -123,7 +136,18 @@ export class MergeChecker {
       const { serviceUrl: serviceURL } = node
       const serviceDomain = extractDomain(serviceURL)
 
-      if (BigInt(nodeTotalDifficulty) === TERMINAL_TOTAL_DIFFICULTY && nodeBlockNumber >= MERGE_BLOCK_NUMBER) {
+      let blockchain = ''
+
+      if (ETHEREUM_BLOCKCHAIN_IDS.includes(blockchainID)) {
+        blockchain = 'ethereum'
+      } else if (GNOSIS_BLOCKCHAIN_IDS.includes(blockchainID)) {
+        blockchain = 'gnosis'
+      }
+
+      if (
```

Comment:
This needs a check of `blockchain` value: we should not look up if this key is not set

Reply:
Also, this should be covered by tests covering a the main scenarios.

Reply:
> This needs a check of `blockchain` value: we should not look up if this key is not set

Addressed in 74aba5b91086c6b9ff77fc8912ae9a136ff04958